SofaKernel/framework/sofa/helper/vector.h

```diff
@@ -184,47 +193,120 @@ class vector<T, CPUMemoryManager<T> > : public std::vector<T, std::allocator<T>
     void fastResize(size_type n) {
         this->resize(n);
     }
+
 };
 
 
+
+
+// ======================  operations on standard vectors
+
+// -----------------------------------------------------------
+//
+/*! @name vector class-related methods
+
+*/
+//
+// -----------------------------------------------------------
+//@{
+/** Remove the first occurence of a given value.
+
+The remaining values are shifted.
+*/
+template<class T1, class T2>
+void remove( T1& v, const T2& elem )
+{
+    typename T1::iterator e = std::find( v.begin(), v.end(), elem );
+    if( e != v.end() )
+    {
+        typename T1::iterator next = e;
+        next++;
+        for( ; next != v.end(); ++e, ++next )
+            *e = *next;
+    }
+    v.pop_back();
+}
+
+/** Remove the first occurence of a given value.
+
+The last value is moved to where the value was found, and the other values are not shifted.
+*/
+template<class T1, class T2>
+void removeValue( T1& v, const T2& elem )
+{
+    typename T1::iterator e = std::find( v.begin(), v.end(), elem );
+    if( e != v.end() )
+    {
+        if (e != v.end()-1)
+            *e = v.back();
```

Comment:
you probably want `*e = std::move(v.back());` here in case `T2` is expensive to copy